### PR TITLE
test: fix failing tests on 4.x LTS branch

### PR DESCRIPTION
**Pull request by FransGH**
### New Pull Request Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [X] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
The CI workflows stalled due to broken tests.
Related issue: #7659
Note: this has been ported to `alpha` via https://github.com/parse-community/parse-server/pull/7667

### Approach
A spec reporter was added helper.js to track completed/failed tests and failed test were repaired.

### TODOs before merging
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: package.json
Comment (mtrezza) [inline package.json]: Is the spec reporter needed to fix the CI?

Comment (FransGH) [reply package.json]: It would have saved substantial time and confusion if it was already included. Not a requirement but definitely helpful!

Comment (mtrezza) [reply package.json]: I guess we can add this, as this is "just" a CI PR. Actually, `alpha` already has this added. Could you downgrade it to version 6.0.0? We want to branches to divert as little as necessary.

**Comment by mtrezza** [comment]
We may also have to port some of the changes in this fix to `alpha`, if they are not already there - could you check? We want to branches to divert as little as necessary.

**Comment by FransGH** [comment]
Wouldn't it make make more sense for alpha to upgrade to jasmine-spec-reporter 7.0.0? 
Happy to create a PR for that, including checking if the other changes in this PR are in the latest master branch.
I'll look into that next week. For now, it would be great if these two PR's could be merged into 4.x.x and released.

**Comment by dblythy** [comment]
Thanks for picking this up @FransGH!!

**Comment by mtrezza** [comment]
> Wouldn't it make make more sense for alpha to upgrade to jasmine-spec-reporter 7.0.0?

Sure, if you want to open another PR, please go ahead. We'll merge this PR together with the one for `alpha` to keep the branches in sync. In the meantime we'll look into how to release a 4.x version, which is on a LTS branch that still has to be set-up for auto-release.

**Comment by FransGH** [comment]
Is there an open issue/pr for 4.x release automation? What are the remaining todo's to get that done?

**Comment by mtrezza** [comment]
@FransGH There is no open GitHub issue for this, but you also don't have to worry about it. I am taking care of this in the background, since it's a more complex undertaking.

To summarize, 4.x is a LTS branch. We can merge this PR if we make sure that it contains only changes that are also in the `alpha` branch, except for changes that are specifically needed for the 4.x branch. So if the jasmine reporter is in the `alpha` branch with different settings or code, we'd need to make sure they are both identical. We'll merge this PR once both PRs are read for merge.

File: package.json
Comment (mtrezza) [reply package.json]: resolve, as https://github.com/parse-community/parse-server/pull/7667 upgrades to 7.0.0

**Comment by mtrezza** [comment]
> In the meantime we'll look into how to release a 4.x version, which is on a LTS branch that still has to be set-up for auto-release.

Since this is just a test fix, there won't be a new release. You would just use the `release-4.x.x` branch for development.

**Review by mtrezza — APPROVED** [review]
Look good! I have made some minor optimizations to the PR to keep it in-sync with the alpha branch, see commit log. Thanks for your contribution!

**Comment by FransGH** [comment]
> Look good! I have made some minor optimizations to the PR to keep it in-sync with the alpha branch, see commit log. Thanks for your contribution!
Happy to be of help. Sorry for my previous holding back for making this simple change, it's just not very compelling making late night changes and not being able to test them as I can't start the ci. But seems that the workflows now run when I make a PR so that's solved too, thanks!

**Comment by mtrezza** [comment]
You did well - we got it merged in the end and that's what counts 😉

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.10.5](https://github.com/parse-community/parse-server/releases/tag/4.10.5)

**Pull request merged**
